working/macros/feature-specification.md

```diff
@@ -246,6 +246,41 @@ introspection APIs.
 These macros can fully introspect on any type reachable from the declarations
 they annotate, including introspecting on members of classes, etc.
 
+## Macro Instantiation
+
+Macros should be able to be able to be instantiated using only the libraries
+that are available to the macro definition library. This allows us to compile an
+app (or spawn an isolate) that can run macro applications for that macro using
+only its own dependencies.
+
+### Macro Constructor Parameters
+
+In order to facilitate this, parameters to Macro constructors are limited to
+a specific set of allowable types:
+
+- [Code][] or its subtypes
+- String
+- int, double, num
+- bool
+- null (really this just allows nullable types)
+- Uri
+- DateTime
+
+### Macro Constructor Arguments
+
+When any type other than [Code][] is used, arguments are only allowed to be
+object literals. Or in the case of `Uri` and `DateTime` they must be direct
```

Comment:
So, the source code for this macro occurs as an annotation in the library using the macro. This *source code* is then executed in a separate isolate. Therefore the source code must be valid in that library. Is that correctly understood?
(And the annotation is not available at run-time in the original library).

In that case, I'd definitely allow any *literal* (including symbol literals, but not including "type literals").

I'm vary about `Uri` and `DateTime` because they do not have const constructors, and therefore shouldn't be in an annotation to begin with. I guess we'll need special rules for macro annotations to allow them to do anything, becaus they'll be instantiated in a non-`const` way eventually. Or you can just put in the string and parse it at the other end.